src/test/java/com/socialmediaraiser/twitterbot/integration/TweetInteractionTest.java

```diff
@@ -29,37 +31,59 @@ public void testAddAnswererId(){
-    assertEquals(1, userInteractions.get("1").get().getNbRetweetsReceived());
-    assertEquals(1, userInteractions.get("1").get().getNbLikesReceived());
-    assertEquals(1, userInteractions.get("2").get().getNbRepliesReceived());
-    assertEquals(0, userInteractions.get("2").get().getNbRetweetsReceived());
-    assertEquals(0, userInteractions.get("2").get().getNbLikesReceived());
-    assertEquals(1, userInteractions.get("3").get().getNbRepliesReceived());
-    assertEquals(1, userInteractions.get("3").get().getNbRetweetsReceived());
-    assertEquals(0, userInteractions.get("3").get().getNbLikesReceived());
-    assertEquals(0, userInteractions.get("4").get().getNbRepliesReceived());
-    assertEquals(1, userInteractions.get("4").get().getNbRetweetsReceived());
-    assertEquals(0, userInteractions.get("4").get().getNbLikesReceived());
-    assertEquals(0, userInteractions.get("5").get().getNbRepliesReceived());
-    assertEquals(0, userInteractions.get("5").get().getNbRetweetsReceived());
-    assertEquals(1, userInteractions.get("5").get().getNbLikesReceived());
-    assertEquals(0, userInteractions.get("6").get().getNbRepliesReceived());
-    assertEquals(0, userInteractions.get("6").get().getNbRetweetsReceived());
-    assertEquals(1, userInteractions.get("6").get().getNbLikesReceived());
+    return TweetInteraction.builder()
+                           .answererIds(answerersIds)
+                           .retweeterIds(retweetIds)
+                           .likersIds(likersIds).build();
   }
 
+  @Test
+  public void testTweetInteractionToUserStats(){
+
+    Map<String, UserStats> result = this.getTweetIntereaction().toUserStatsMap();
+
+    assertEquals(1, result.get("1").get().getNbRepliesReceived());
+    assertEquals(1, result.get("1").get().getNbRetweetsReceived());
+    assertEquals(1, result.get("1").get().getNbLikesReceived());
+    assertEquals(1, result.get("2").get().getNbRepliesReceived());
+    assertEquals(0, result.get("2").get().getNbRetweetsReceived());
+    assertEquals(0, result.get("2").get().getNbLikesReceived());
+    assertEquals(1, result.get("3").get().getNbRepliesReceived());
+    assertEquals(1, result.get("3").get().getNbRetweetsReceived());
+    assertEquals(0, result.get("3").get().getNbLikesReceived());
+    assertEquals(0, result.get("4").get().getNbRepliesReceived());
+    assertEquals(1, result.get("4").get().getNbRetweetsReceived());
+    assertEquals(0, result.get("4").get().getNbLikesReceived());
+    assertEquals(0, result.get("5").get().getNbRepliesReceived());
+    assertEquals(0, result.get("5").get().getNbRetweetsReceived());
+    assertEquals(1, result.get("5").get().getNbLikesReceived());
+    assertEquals(0, result.get("6").get().getNbRepliesReceived());
+    assertEquals(0, result.get("6").get().getNbRetweetsReceived());
+    assertEquals(1, result.get("6").get().getNbLikesReceived());
+  }
 
+  @Test
+  public void testMapOfTweetInteractionToMapOfUserStats(){
```

Comment:
@hypr2771 if you have time to waste, i created the method `toUserStatsMap` in `TweetInteraction` and this unit test but it looks like the `.collect` you gave me is not working as expected :)